### PR TITLE
Build with lts-16.31 / ghc-8.8.4

### PR DESCRIPTION
**Pull request by pbrisbin**
This only required a minor loosening of bounds.

I attempted to go further (we include down to lts-12 when we can), but I
hit a dependency-cycle error (aeson, these, aeson, monad-logger-aeson).
I've not seen this before and don't really know how to trouble-shoot.

### CODE REVIEW

**Comment by pbrisbin** [comment]
Is it intentional that your CI doesn't run on PRs?

On my own pushes I've got this passing except the OSX build. Not sure I can figure that one out, is it a deal breaker?

File: monad-logger-aeson/package.yaml [hunk 1]
```diff
@@ -48,12 +48,12 @@ tests:
...
-    - aeson-diff >=1.1.0.10 && <1.2.0.0
-    - base >=4.14.0.0 && <5
-    - bytestring >=0.10.12.0 && <0.12.0.0
+    - aeson >=1.5.2.0 && <2.1.3.0
+    - aeson-diff >=1.1.0.9 && <1.2.0.0
+    - base >=4.13.0.0 && <5
+    - bytestring >=0.10.10.1 && <0.12.0.0
     - directory >=1.3.6.0 && <1.4.0.0
-    - hspec >=2.7.10 && <2.10.0
+    - hspec >=2.7.6 && <2.10.0
```
Comment (jship) [inline monad-logger-aeson/package.yaml]: Let's change this lower bound from `2.7.6` to `2.7.9` so that it matches `stack-lts-16.31.yaml`.

**Review by jship — CHANGES_REQUESTED** [review]
Thanks for doing this! I requested a small change for the `hspec` lower bound and then it'll be all set to merge.

**Comment by jship** [comment]
> Is it intentional that your CI doesn't run on PRs?

Nah, was mostly just that it wasn't needed while the library was in it's initial development phase. I can turn it on after this PR.

> On my own pushes I've got this passing except the OSX build. Not sure I can figure that one out, is it a deal breaker?

Np, we can exclude that one in the build matrix. Would you be interested in adding that exclusion to `.github/workflows/ci.yaml`?

**Comment by jship** [comment]
> > On my own pushes I've got this passing except the OSX build. Not sure I can figure that one out, is it a deal breaker?
> 
> Np, we can exclude that one in the build matrix. Would you be interested in adding that exclusion to `.github/workflows/ci.yaml`?

Ah, taking a closer look at https://github.com/pbrisbin/monad-logger-aeson/runs/6758786113?check_suite_focus=true, it sounds like there is something screwy going on when build artifacts are present in the cache from newer GHCs on older GHCs ([ref 1](https://gitlab.haskell.org/ghc/ghc/-/issues/19452) [ref 2](https://gitlab.haskell.org/ghc/ghc/-/issues/18228)). The cache step is picking up the cache with key `macOS-lts-19.6` which would have GHC 9.0.2 build artifacts.

Perhaps the cleanest way around this is to not use `restore-keys` at all, so that if we have a cache miss, we don't try to restore from other caches. Would you mind trying out whacking this line? https://github.com/jship/monad-logger-aeson/blob/268ece4d43bd59d727e0515c2f47980d42042224/.github/workflows/ci.yaml#L31

File: monad-logger-aeson/package.yaml (hunk 1 above)
Comment (pbrisbin) [reply monad-logger-aeson/package.yaml]: Ah good catch.

**Comment by pbrisbin** [comment]
> Perhaps the cleanest way around this is to not use restore-keys at all

I'm happy to do whatever you like, but we maintain [an action](https://github.com/freckle/stack-cache-action#readme) that handles this. It incorporates `os.runner` and `stack-yaml` into the key and so should avoid this issue. We've not tested it on your setup of `stack.yaml` as a symlink though, so I'm not sure if that'd confuse it.

Would you like me to try it, or just remove the `restore-keys`?

**Review by jship — COMMENTED** [review]
Change looks good. Would you check in the regenerated cabal file too?

**Comment by jship** [comment]
> > Perhaps the cleanest way around this is to not use restore-keys at all
> 
> I'm happy to do whatever you like, but we maintain [an action](https://github.com/freckle/stack-cache-action#readme) that handles this. It incorporates `os.runner` and `stack-yaml` into the key and so should avoid this issue. We've not tested it on your setup of `stack.yaml` as a symlink though, so I'm not sure if that'd confuse it.
> 
> Would you like me to try it, or just remove the `restore-keys`?

Ah that action looks really nice. Trying that out sounds great!

As for the symlink, that's only there for convenience of local dev. It points at `stack-nightly-2022-05-05.yaml`. `ci.yaml` references the concrete stack yamls, so I believe we'd be all set there if we used the `stack-yaml` input to Freckle's `stack-cache-action`.

**Comment by pbrisbin** [comment]
It seems we've hit an issue with caching on OSX. We've fought with this before, something about the BSD tar implementation producing corrupt files. I'm going to poke at it a little, I'll let you know if/when I figure it out.

**Comment by pbrisbin** [comment]
OK, I think this is good now. I had to update the `stack-cache-action` and then trigger the _save_ side of it on the new version, which I did by adding a `prefix`. That value doesn't matter and could be removed once no caches exist that were created by the old un-prefixed runs -- if you want.

Let me know if you'd like me to clean up history, or do anything else.

**Review by jship — APPROVED** [review]
Thanks so much! Looks great.

**Comment by jship** [comment]
I'll get a new version out on hackage with these changes before this week wraps up.

**Comment by jship** [comment]
The `0.2.0.1` version including these changes is now on [Hackage](https://hackage.haskell.org/package/monad-logger-aeson-0.2.0.1).

**Pull request merged**